### ISSUE TITLE
Rapid rebind leads to an address in use exception

### ISSUE THREAD

**Issue by mrakgr** [zeromq/netmq#898]
### Environment

    NetMQ Version:    4.0.1.2-pre
    Operating System: Windows 10
    .NET Version:     .NET Core 3.1

### Expected behaviour

The example syncs with all 10 clients.

### Actual behaviour

Get an address in use exception after the first sync. With minimal thread stack size I get a stack overflow instead.

### Steps to reproduce the behaviour

Run [the example](https://github.com/mrakgr/Lithe-POC/blob/7c6d1833b2b0fa9dc4796112f068b1acdee07cc7/Lithe%20Avalonia%20Examples/ZeroMQ_WeatherSyncedBug.fs).

---

```fsharp
        let server (log : string -> unit) (poller : NetMQPoller) =
            try let rand = Random()
                init PublisherSocket poller (bind uri) <| fun pub ->
                log <| sprintf "Publisher has bound to %s." uri
                //init ResponseSocket poller (bind uri_start) <| fun initer ->
                //    for i=1 to num_subs do initer.ReceiveFrameString() |> ignore; initer.SendFrameEmpty()
                for i=1 to num_subs do ResponseSocket.sync_receive_string uri_start |> ignore
```

These last few lines are the ones that are interesting. The way `ResponseSocket.sync_receive_string` is implemented is...

```fsharp
    module ResponseSocket =
        let inline sync_receive_string uri =
            use s = new ResponseSocket()
            bind uri s <| fun () ->
            let r = s.ReceiveFrameString()
            s.SendFrameEmpty()
            r
```

What the loop does is create a new response socket, binds to the uri, receives a string, sends an empty frame back and then returns the string.

`bind uri s <| fun () ->`

This particular helper just binds the socket and then unbinds it when it goes out of scope, similar to an `use` statement.

```fsharp
                //init ResponseSocket poller (bind uri_start) <| fun initer ->
                //    for i=1 to num_subs do initer.ReceiveFrameString() |> ignore; initer.SendFrameEmpty()
```

This commented out version where the socket is maintained for the duration of the sync is what works.

In the real world, I'd use NetMQ servers for the duration for the process so I would not run into these kinds of issues, but nonetheless I think these are genuine bugs you might be interested in fixing.

This example is my translation from the [Node Coordination section](http://zguide.zeromq.org/page:all#Node-Coordination).

**Comment by mrakgr** [comment]
Thinking about it a bit more closely, this error might be on my end. I don't know if the blocking send actually fills up some queue for the receiver on the other end. If it does then this issue is not really a bug, but otherwise it should be.

**Comment by somdoron** [comment]
There is a notimewait option on a socket, try to set it. Also, calling unbind before dispose might be faster

**Comment by mrakgr** [comment]
> There is a notimewait option on a socket, try to set it.

Still get an exception that the socket is in use when I do that. I am not sure if I explained it properly, but `for i=1 to num_subs do ResponseSocket.sync_receive_string uri_start |> ignore` is intended to create and dispose `num_subs` (in this case 10) sockets in rapid succession.

> Also, calling unbind before dispose might be faster

That is what I do in all the examples.

`bind uri s <| fun () ->`

This `bind` is a helper that I've made so that everything after it is wrapped in a lambda, similar to an `use` expression. It unbinds automatically after the lambda is executed.
